src/engine/strat_engine/thinpool.rs

```diff
@@ -89,46 +149,92 @@ impl ThinPool {
     /// If initial setup fails due to a thin_check failure, attempt to fix
     /// the problem by running thin_repair. If failure recurs, return an
     /// error.
-    #[allow(too_many_arguments)]
     pub fn setup(pool_uuid: PoolUuid,
                  dm: &DM,
                  data_block_size: Sectors,
                  low_water_mark: DataBlocks,
-                 thin_ids: &[ThinDevId],
-                 spare_segments: Vec<Segment>,
-                 meta_dev: LinearDev,
-                 data_dev: LinearDev,
-                 mdv: MetadataVol,
-                 fs_save: Vec<FilesystemSave>)
+                 flex_devs: &FlexDevsSave,
+                 bd_mgr: &BlockDevMgr)
                  -> EngineResult<ThinPool> {
+
+        let mapper = |triple: &(Uuid, Sectors, Sectors)| -> EngineResult<BlkDevSegment> {
```

Comment:
I think you should call a function on ```bd_mgr``` that returns a mapper function, as I do in my addendum to previous PR, rather than use ```get_by_uuid()``` to construct this mapper function. Then ```BlockDevMgr``` has control over how mapping is constructed, and efficiency is also gained.